app/backend/src/couchers/servicers/requests.py

```diff
@@ -584,3 +592,75 @@ def MarkLastSeenHostRequest(self, request, context):
 
             session.commit()
             return empty_pb2.Empty()
+
+    def GetResponseRate(self, request, context):
+        with session_scope() as session:
+            # this subquery gets the time that the request was sent
+            t = (
+                select(Message.conversation_id, Message.time)
+                .where(Message.message_type == MessageType.chat_created)
+                .subquery()
+            )
+            # this subquery gets the time that the user responded to the request
+            s = (
+                select(Message.conversation_id, func.min(Message.time).label("time"))
+                .where(Message.author_id == request.user_id)
+                .group_by(Message.conversation_id)
+                .subquery()
+            )
+
+            res = session.execute(
+                select(
+                    User.id,
+                    func.count().label("n"),
+                    func.count(s.c.time) / (1.0 * func.greatest(func.count(t.c.time), 1)).label("response_rate"),
+                    percentile_disc(0.33)
+                    .within_group(func.coalesce(s.c.time - t.c.time, timedelta(days=1000)))
+                    .label("response_time_p33"),
+                    percentile_disc(0.66)
+                    .within_group(func.coalesce(s.c.time - t.c.time, timedelta(days=1000)))
+                    .label("response_time_p66"),
```

Comment:
Ah, I see. Though I do think they're good for maintainability? I can of course convert them to comments too

Reply:
There's no automated checking that they correspond to the variable names when you unpack the result. So better use comments so that we don't get a false sense of some automated checks.